src/compiler/scala/tools/nsc/typechecker/SyntheticMethods.scala

```diff
@@ -318,14 +318,15 @@ trait SyntheticMethods extends ast.TreeDSL {
 
     /* If you serialize a singleton and then deserialize it twice,
      * you will have two instances of your singleton unless you implement
-     * readResolve.  Here it is implemented for all objects which have
-     * no implementation and which are marked serializable (which is true
-     * for all case objects.)
+     * readResolve.  Here we use a serialization proxy that prevents
```

Comment:
I don't think the comment is quite right, how about 

/* If you serialize a singleton and then deserialize it twice, 
 * you will have two instances of your singleton unless you implement 
 * special serialization logic.  Here we use a serialization proxy that prevents 
 * serialization of state and will, on deserialization by replaced by the object
 * via use of readResolve. This is done for all top level objects which extend
 * `java.io.Serializable` (such as case objects)
 */